frontend/src/metabase/static-viz/components/XYChart/types.ts

```diff
@@ -53,6 +69,7 @@ export type ChartSettings = {
     bottom?: string;
     right?: string;
   };
+  series_settings?: Record<string, { color?: string; title?: string }>;
```

Comment:
this new setting will be present when the column color is set. e.g.
![image](https://user-images.githubusercontent.com/1937582/201140305-9ebf3b54-2958-4c25-a216-03b363f88687.png)

Or when a custom column name is set.